tests/test_acmpca/test_acmpca.py

```diff
@@ -697,3 +696,84 @@ def test_policy_operations():
     with pytest.raises(ClientError) as exc:
         client.get_policy(ResourceArn=ca_arn)
     assert exc.value.response["Error"]["Code"] == "ResourceNotFoundException"
+
+
+@mock_aws
+def test_revoke_certificate():
+    client = boto3.client("acm-pca", region_name="us-east-1")
+
+    # Create and activate a CA
+    ca_arn = client.create_certificate_authority(
+        CertificateAuthorityConfiguration={
+            "KeyAlgorithm": "RSA_4096",
+            "SigningAlgorithm": "SHA512WITHRSA",
+            "Subject": {"CommonName": "test.example.com"},
+        },
+        CertificateAuthorityType="ROOT",
+    )["CertificateAuthorityArn"]
+
+    # Get CSR and issue certificate to activate the CA
+    csr = client.get_certificate_authority_csr(CertificateAuthorityArn=ca_arn)["Csr"]
+    certificate_arn = client.issue_certificate(
+        CertificateAuthorityArn=ca_arn,
+        Csr=csr,
+        SigningAlgorithm="SHA256WITHRSA",
+        TemplateArn="arn:aws:acm-pca:::template/RootCACertificate/V1",
+        Validity={"Type": "YEARS", "Value": 10},
+    )["CertificateArn"]
+    cert = client.get_certificate(
+        CertificateAuthorityArn=ca_arn, CertificateArn=certificate_arn
+    )["Certificate"]
+    client.import_certificate_authority_certificate(
+        CertificateAuthorityArn=ca_arn,
+        Certificate=cert,
+    )
+
+    # Issue a certificate that we'll later revoke
+    test_csr = client.get_certificate_authority_csr(CertificateAuthorityArn=ca_arn)[
+        "Csr"
+    ]
+    test_cert_arn = client.issue_certificate(
+        CertificateAuthorityArn=ca_arn,
+        Csr=test_csr,
+        SigningAlgorithm="SHA256WITHRSA",
+        Validity={"Type": "YEARS", "Value": 1},
+    )["CertificateArn"]
+
+    # Get the certificate before revocation - this should work
+    test_cert_response = client.get_certificate(
+        CertificateAuthorityArn=ca_arn, CertificateArn=test_cert_arn
+    )
+    assert "Certificate" in test_cert_response
+    test_cert = test_cert_response["Certificate"]
+
+    # Get the serial number from the certificate
+    cert_obj = cryptography.x509.load_pem_x509_certificate(test_cert.encode("utf-8"))
+    serial_number = format(cert_obj.serial_number, "X")
+    serial_number = ":".join(
+        serial_number[i : i + 2] for i in range(0, len(serial_number), 2)
+    )
+
+    # Revoke certificate
+    client.revoke_certificate(
+        CertificateAuthorityArn=ca_arn,
+        CertificateSerial=serial_number,
+        RevocationReason="KEY_COMPROMISE",
+    )
+
+    # Verify the certificate is revoked by checking the backend directly
```

Comment:
This check fails in ServerMode unfortunately, because we can't reach the backend when Moto is running in a separate process. 

The next check (that `get_certificate` fails) should work, so ideally we only guard this section (`if TEST_SERVER_MODE: ...`)

Reply:
Seems like that isn't working out on the proxy mode. I have added a skip for this whole test in proxy mode. Let me know if that's not ideal!

Reply:
Makes sense @HarshCasper! Looks like there are just some merge conflicts, once that's fixed I'd be happy to merge